build-frontend/action.yml

```diff
@@ -37,58 +37,57 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - name: Setup node
-      uses: actions/setup-node@v3
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to Docker Registry
+      uses: docker/login-action@v3
       with:
-        node-version: 18.x
+        registry: ${{ inputs.docker_registry_url }}
+        username: ${{ inputs.cloud_username }}
+        password: ${{ inputs.cloud_password }}
 
-    - name: Encode Basic Auth
-      id: base64-auth
+    - name: Resolve values of the versions, docker ref & dockerfile path
+      id: resolve
       shell: bash
       run: |
-        echo "BASIC_AUTH_TOKEN=$(echo -n ${{ inputs.cloud_username }}:${{ inputs.cloud_password }} | base64 -w 0)" >> "$GITHUB_OUTPUT"
+        echo "version=${{ inputs.version || github.sha }}" >> $GITHUB_OUTPUT
+        echo "ref=${{ inputs.docker_registry_url }}/${{ inputs.project_name }}-storefrontcloud-io" >> $GITHUB_OUTPUT
+        if [[ $FRONTEND == "next" ]]; then
+          echo "path=.vuestorefrontcloud/docker/nextjs/Dockerfile-frontend" >> $GITHUB_OUTPUT
+        elif [[ $FRONTEND == "nextjs" ]]; then
+          echo "path=.vuestorefrontcloud/docker/nextjs/Dockerfile-frontend" >> $GITHUB_OUTPUT
+        elif [[ $FRONTEND == "nuxt" ]]; then
+          echo "path=.vuestorefrontcloud/docker/nuxtjs/Dockerfile-frontend" >> $GITHUB_OUTPUT
```

Comment:
Can we add comments explaining what's the GITHUB_OPTIONS var is being set here?
I know it was already here before, but looks like you've been touching the flow here so you probably know what's going on here

Reply:
Good point 👍